core/src/main/java/io/confluent/rest/SslFactory.java

```diff
@@ -31,14 +34,40 @@ public final class SslFactory {
 
   private SslFactory() {}
 
+  private static boolean useBcfks(String provider) {
+    if(provider == null) {
+      return false;
+    }
+    return provider.equals(SslConfigs.FIPS_SSL_PROVIDER);
```

Comment:
this function can be simplify as Objects.equals(SslConfigs.FIPS_SSL_PROVIDER, provider), because SslConfigs.FIPS_SSL_PROVIDER is not null.